---
title: "RNASeq"
author: "m2407447"
output:
  html_document: default
  pdf_document: default
---

```{r load-data-counts, message=FALSE, warning=FALSE}
# Load the required libraries
library("DESeq2")
library("ggplot2")

# Load the count data
counts <- read.csv("C:/Users/Asus/OneDrive/Desktop/RStudio assessment/RNAseq-assessment/exercise1_counts.csv", 
                   row.names = 1)

# Load the sample description
sample_description <- read.table("C:/Users/Asus/OneDrive/Desktop/RStudio assessment/RNAseq-assessment/exercise1_sample_description.info", 
                                 header = TRUE, sep = "\t", row.names = 1)

# Set rownames of sample_description to match colnames of counts
rownames(sample_description) <- colnames(counts)

# Check alignment between colnames and rownames
all(colnames(counts) == rownames(sample_description))
```
```{r}
# Fix missing 'condition' values
sample_description[c("control_FFa1.bam", "control_FFa2.bam", "control_FFa3.bam"), "condition"] <- "control"
sample_description[c("mutant_KOa1.bam", "mutant_KOa2.bam", "mutant_KOa3.bam"), "condition"] <- "KOa"
sample_description[c("mutant_KOb1.bam", "mutant_KOb2.bam", "mutant_KOb3.bam"), "condition"] <- "KOb"

# Ensure 'condition' column is a factor
sample_description$condition <- factor(sample_description$condition, levels = c("control", "KOa", "KOb"))

# Construct DESeqDataSet
dds <- DESeqDataSetFromMatrix(countData = counts, 
                              colData = sample_description, 
                              design = ~ condition)

# Inspect the DESeqDataSet
dds
```

```{r run-deseq, message=FALSE, warning=FALSE}
# Run DESeq2 differential expression analysis
dds <- DESeq(dds)
```
```{r}
# Get results
res <- results(dds)

# View top results
head(res)
```



